src/rdata.cpp

```diff
@@ -833,6 +854,14 @@ void ReturnData::fsres(const int it, Model &model, const ExpData &edata) {
             sres.at(idx) = amici::fsres(y_it.at(iy), sy_it.at(iy + ny * ip),
                                         observedData[iy], sigmay_it.at(iy),
                                         ssigmay_it.at(iy + ny * ip));
+            if (sigma_res) {
+                int idx_res =
+                    (iy + it * edata.nytrue() + edata.nytrue() * edata.nt()) *
```

Comment:
cache result of `edata.n*()` outside nested loop? Could also avoid some repeated `.at()`s, but not sure how relevant.

Reply:
as those are not called during simulation, computational cost is pretty negligible here.